plutus-contract/test/Spec/Contract/Tx/Constraints/MustProduceAtLeast.hs

```diff
@@ -79,10 +90,9 @@ mustProduceAtLeastContract offAmt onAmt baseScriptValue pkh = do
     let lookups2 = Constraints.typedValidatorLookups typedValidator
             <> Constraints.unspentOutputs pubKeyUtxos
             <> Constraints.unspentOutputs scriptUtxos
-            <> Constraints.ownPaymentPubKeyHash pkh
         tx2 =
             Constraints.collectFromTheScript scriptUtxos ()
-            <> Constraints.mustIncludeDatumInTx (Datum $ PlutusTx.toBuiltinData onAmt)
+            <> Constraints.mustPayWithDatumInTxToPubKey w1PaymentPubKeyHash (Datum $ PlutusTx.toBuiltinData onAmt) offAmt
```

Comment:
Let's do everything with `Address` from now on (unless you're testing one of the pubkey functions specifically).